modules/aggregation-layers/src/contour-layer/contour-layer.js

```diff
@@ -37,160 +37,158 @@ const defaultProps = {
+  aggregation: 'SUM',
 
   // contour lines
   contours: [{threshold: DEFAULT_THRESHOLD}],
 
-  fp64: false,
   zOffset: 0.005
 };
 
 // props , when changed requires re-aggregation
-const AGGREGATION_PROPS = ['gpuAggregation'];
+const AGGREGATION_PROPS = ['aggregation', 'getWeight'];
 
 export default class ContourLayer extends GridAggregationLayer {
   initializeState() {
-    super.initializeState(AGGREGATION_PROPS);
+    super.initializeState({aggregationProps: AGGREGATION_PROPS});
     this.setState({
       contourData: {},
-      colorTrigger: 0,
-      strokeWidthTrigger: 0
+      weights: {
+        count: {
+          size: 1,
+          operation: AGGREGATION_OPERATION.SUM
+        }
+      }
     });
     const attributeManager = this.getAttributeManager();
     attributeManager.add({
-      positions: {size: 3, accessor: 'getPosition', type: GL.DOUBLE, fp64: false},
+      positions: {size: 3, accessor: 'getPosition'},
+      // this attribute is used in gpu aggregation path only
       count: {size: 3, accessor: 'getWeight'}
     });
   }
 
   updateState(opts) {
     super.updateState(opts);
     let contoursChanged = false;
-    const dataChanged = this._isAggregationDirty(opts);
-    const cellSizeChanged = opts.oldProps.cellSize !== opts.props.cellSize;
-    if (dataChanged || cellSizeChanged) {
-      this.setState({countsData: null});
-      this._aggregateData({
-        dataChanged,
-        cellSizeChanged
-      });
-    }
-    if (this._shouldRebuildContours(opts)) {
+    const {oldProps, props} = opts;
+    const {aggregationDirty} = this.state;
+
+    if (oldProps.contours !== props.contours || oldProps.zOffset !== props.zOffset) {
       contoursChanged = true;
       this._updateThresholdData(opts.props);
     }
-    if (dataChanged || cellSizeChanged || contoursChanged) {
+
+    if (this.getNumInstances() > 0 && (aggregationDirty || contoursChanged)) {
       this._generateContours();
-    } else {
-      // data for sublayers not changed check if color or strokeWidth need to be updated
-      this._updateSubLayerTriggers(opts.oldProps, opts.props);
     }
   }
 
   renderLayers() {
     const {contourSegments, contourPolygons} = this.state.contourData;
-    const hasIsolines = contourSegments && contourSegments.length > 0;
-    const hasIsobands = contourPolygons && contourPolygons.length > 0;
 
-    const lineLayer = hasIsolines && new LineLayer(this._getLineLayerProps());
-    const solidPolygonLayer =
-      hasIsobands && new SolidPolygonLayer(this._getSolidPolygonLayerProps());
-    return [lineLayer, solidPolygonLayer];
+    const LinesSubLayerClass = this.getSubLayerClass('lines', LineLayer);
+    const BandsSubLayerClass = this.getSubLayerClass('bands', SolidPolygonLayer);
+
+    // Contour lines layer
+    const lineLayer =
+      contourSegments &&
+      contourSegments.length > 0 &&
+      new LinesSubLayerClass(
+        this.getSubLayerProps({
+          id: 'contour-line-layer'
```

Comment:
- Just use sub layer ids "lines" and "bands". They will be concatenated with the parent layer's id so you don't have to call out that they are children of a contour layer.
- The sub layer ids should be documented in `contour-layer.md`